public/components/data_connections/components/access_control_tab.tsx

```diff
@@ -0,0 +1,205 @@
+                  : '-'}
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem grow={false}>
+              <EuiText className="overview-title">Acceleration permissions</EuiText>
+              <EuiText size="s" className="overview-content">
+                {finalAccelerationPermissionsPlaceHolder.length
+                  ? `Restricted to ${finalAccelerationPermissionsPlaceHolder.map(
+                      (role) => role.label
+                    )}`
+                  : '-'}
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
+  const renderEditAccessControlDetails = () => {
+    return (
+      <EuiFlexGroup direction="column">
+        <QueryPermissionsFlexItem
+          roles={roles}
+          selectedRoles={selectedQueryPermissionRoles}
+          setSelectedRoles={setSelectedQueryPermissionRoles}
+          selectedRadio={queryPermissionRadioSelected}
+          setSelectedRadio={setQueryRadioIdSelected}
+          radios={queryRadios}
+        />
+        <AccelerationPermissionsFlexItem
+          roles={roles}
+          selectedRoles={selectedAccelerationPermissionRoles}
+          setSelectedRoles={setSelectedAccelerationPermissionRoles}
+          selectedRadio={accelerationPermissionRadioSelected}
+          setSelectedRadio={setAccelerationRadioIdSelected}
+          radios={accelerationRadios}
+        />
+      </EuiFlexGroup>
+    );
+  };
+
+  const saveChanges = () => {
+    setFinalAccelerationPermissionsPlaceHolder(selectedAccelerationPermissionRoles);
+    setFinalQueryPermissionsPlaceHolder(selectedQueryPermissionRoles);
+    setMode('view');
+  };
+
+  return (
+    <>
+      <EuiSpacer />
+      <AccessControlCallout />
+      <EuiSpacer />
+      <EuiPanel>
+        <EuiFlexGroup direction="row">
+          <EuiFlexItem>
+            <EuiText size="m">
+              <h2 className="panel-title">Access Control</h2>
+              Control which OpenSearch users have access to this data source.
+            </EuiText>
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              data-test-subj="createButton"
+              onClick={
+                mode === 'view'
+                  ? () => {
+                      setMode('edit');
+                    }
+                  : () => {
+                      setMode('view');
+                    }
+              }
+              fill={mode === 'view' ? true : false}
+            >
+              {mode === 'view' ? 'Edit' : 'Cancel'}
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiHorizontalRule />
+        {mode === 'view' ? renderViewAccessControlDetails() : renderEditAccessControlDetails()}
```

Comment:
Consider setting this value to a component-constant (`AccessControlDetails`).  You have several of these conditions - they can all be collected into one place... thereby reducing cyclomatic complexity.

Containing these conditions into one location will also raise the visibility of refactoring / simplification opportunities.

Reply:
Not quite sure I understand what you are suggesting/how it would be cleaner than having a state variable. Can you point me to some examples I can look at?